astropy/io/fits/hdu/compressed.py

```diff
@@ -1574,6 +1574,34 @@ def data(self, data):
                 )
             )
 
+    def data_astype(self, data_type):
```

Comment:
My sense would be to use this function for the property, i.e., here do `data_type='numpy'` and then have `data = property(data_astype)`.

I think you can still do `@data.setter` after, though it may be an idea to already know make this a 2-way process, i.e., have a `set_data()` (class) method. But probably fine to postpone that (though for my `baseband` package, I have found that writing converters in two directions really helps to get rid of bugs!).